src/transaction_signature_options/sign_with_mpc/mod.rs

```diff
@@ -0,0 +1,885 @@
+                            network_config,
+                            &sign_request_tx,
+                        ) {
+                            Ok(outcome_view) => outcome_view,
+                            Err(error) => return Err(error),
+                        };
+
+                    let signed_transaction = match sign_outcome_view.status {
+                        near_primitives::views::FinalExecutionStatus::SuccessValue(result) => {
+                            let sign_result: mpc_sign_result::SignResult =
+                                serde_json::from_slice(&result)?;
+                            let signature: near_crypto::Signature = sign_result.into();
+
+                            near_primitives::transaction::SignedTransaction::new(
+                                signature,
+                                unsigned_transaction,
+                            )
+                        }
+                        _ => {
+                            let error_msg = format!("Failed to sign MPC transaction for <{sender_id}>\nUnexpected outcome view after sending to \"sign\" to <{contract_id}> contract.");
+                            eprintln!("{error_msg}");
+                            return Err(color_eyre::eyre::eyre!(error_msg));
+                        }
+                    };
+
+                    tracing::info!(
+                        parent: &tracing::Span::none(),
+                        "Successfully signed original transaction from <{}> to <{}> via MPC contract <{}>:{}",
+                        sender_id,
+                        receiver_id,
+                        contract_id,
+                        crate::common::indent_payload(&format!(
+                            "\nSignature:  {}\n",
+                            signed_transaction.signature,
+                        ))
+                    );
+
+                    *signed_transaction_to_replace = signed_transaction;
+
+                    Ok(())
+                }
+            });
+
+        let on_after_sending_transaction_callback: crate::transaction_signature_options::OnAfterSendingTransactionCallback = std::sync::Arc::new({
+            move |outcome_view, network_config| {
+                let global_context = global_context_for_after_send.clone();
+                let unsigned_transaction = original_transaction_for_after_send.clone();
+                let mpc_sign_request = item.mpc_sign_request.clone();
+
+                // NOTE: checking if outcome view status is not failure is not necessary, as this
+                // callback will be called only after `crate::common::print_transaction_status`,
+                // which will check for failure of transaction already
+
+                if let Some(near_primitives::views::ActionView::FunctionCall { method_name, args, .. }) =
+      outcome_view.transaction.actions.first()
+                {
+                    if method_name == "add_proposal" {
```

Comment:
I am concerned about multiple places which rely on the method name to be `add_proposal`. Can you capture the necessary data into the closure instead of parsing it back out of the transaction params?